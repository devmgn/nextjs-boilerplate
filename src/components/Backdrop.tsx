'use client';

import styled from 'styled-components';
import createShouldForwardProp from '@/utils/createShouldForwardProp';

const Backdrop = styled.div
  .withConfig({
    shouldForwardProp: createShouldForwardProp('transparent'),
  })
  .attrs({ className: 'Backdrop', 'aria-hidden': true })`
  position: fixed;
  inset: 0;
  background-color: rgba(0, 0, 0, 0.5);
`;

export default Backdrop;
