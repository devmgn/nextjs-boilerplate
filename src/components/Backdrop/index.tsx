import { Presence } from "@radix-ui/react-presence";
import { tv } from "tailwind-variants";
import type { VariantProps } from "tailwind-variants";

const backdropVariants = tv({
  base: "fixed inset-0 select-none bg-black/50 data-[state=closed]:animate-[fade-out_0.2s_ease-in-out] data-[state=open]:animate-[fade-in_0.2s_ease-in-out]",
  variants: {
    blur: {
      true: "backdrop-blur-xs",
    },
  },
  defaultVariants: {
    blur: true,
  },
});

interface BackdropProps
  extends React.ComponentProps<"div">,
    VariantProps<typeof backdropVariants> {
  open: boolean;
}

export function Backdrop(props: BackdropProps) {
  const { open, blur, className, ...restProps } = props;

  return (
    <Presence present={open}>
      <div
        {...restProps}
        aria-hidden="true"
        className={backdropVariants({ className, blur })}
        data-state={open ? "open" : "closed"}
        role="presentation"
      />
    </Presence>
  );
}
