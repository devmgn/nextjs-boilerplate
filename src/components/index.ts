export * from './Footer';
export * from './Header';
export * from './ErrorFallbackRender';
export * from './Hydrator';
