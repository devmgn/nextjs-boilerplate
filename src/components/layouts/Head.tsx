import React from 'react';
import { default as NextHead } from 'next/head';
import appSettings from 'appSettings';

type HeadProps = {
  title?: string;
  description?: string;
  children?: React.ReactNode;
};

const Head: React.FC<HeadProps> = (props) => {
  const { title, description } = props;
  const { siteName, titleSeparator } = appSettings;

  return (
    <NextHead>
      <title>{title ? `${title}${titleSeparator}${siteName}` : siteName}</title>
      {description && <meta name="description" content={description} />}
      {props.children}
    </NextHead>
  );
};

export default Head;
