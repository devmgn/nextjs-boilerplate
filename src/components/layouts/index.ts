export { default as Footer } from '@/components/layouts/Footer';
export { default as Layout } from '@/components/layouts/Layout';
export { default as Header } from '@/components/layouts/Header';
export { default as Head } from '@/components/layouts/Head';
