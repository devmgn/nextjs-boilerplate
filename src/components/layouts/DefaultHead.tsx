import Head from 'next/head';
import appSettings from 'appSettings';

const DefaultHead = () => {
  return (
    <Head>
      <meta charSet="utf-8" />
      <meta name="viewport" content="width=device-width,initial-scale=1.0,minimum-scale=1.0" />
      <title>{appSettings.meta.appName}</title>
      <meta name="description" content={appSettings.meta.defaultDescription} />
      <link rel="icon" href="/images/favicon.ico" />
    </Head>
  );
};

export default DefaultHead;
