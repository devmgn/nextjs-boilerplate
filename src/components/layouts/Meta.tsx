import Head from 'next/head';
import appSettings from 'appSettings';

type HeadProps = {
  title?: string;
  description?: string;
  children?: React.ReactNode;
};

const {
  meta: { appName, defaultDescription, titleSeparator },
} = appSettings;

const Meta: React.FC<HeadProps> = ({ title, description, children }) => {
  return (
    <Head>
      <title>{title ? `${title}${titleSeparator}${appName}` : appName}</title>
      <meta name="description" content={description ?? defaultDescription} />
      {children}
    </Head>
  );
};

export default Meta;
