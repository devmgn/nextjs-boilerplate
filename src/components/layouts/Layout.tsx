import Head from 'next/head';
import { Header, Footer } from '@/components/layouts';
import appSettings from 'appSettings';

export type LayoutProps = {
  children: React.ReactNode;
};

const Layout: React.FC<LayoutProps> = (props) => {
  return (
    <>
      <Head>
        <meta charSet="utf-8" />
        <meta name="viewport" content="width=device-width,initial-scale=1.0,minimum-scale=1.0" />
        <title>{appSettings.siteName}</title>
        <meta name="description" content={appSettings.defaultDescription} />
        <link rel="icon" href="/images/favicon.ico" />
      </Head>
      <Header />
      <main>{props.children}</main>
      <Footer />
    </>
  );
};

export default Layout;
