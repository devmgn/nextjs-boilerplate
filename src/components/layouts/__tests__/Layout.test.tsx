import { render } from '@testing-library/react';
import { Layout } from '@/components/layouts';

describe('Layout', () => {
  test('スナップショットテスト', () => {
    const { asFragment } = render(<Layout>children</Layout>);
    expect(asFragment()).toMatchSnapshot();
  });
});
