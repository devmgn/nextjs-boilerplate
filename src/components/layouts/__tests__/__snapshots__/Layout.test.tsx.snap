// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout スナップショットテスト 1`] = `
<DocumentFragment>
  <header>
    header
  </header>
  <main>
    children
  </main>
  <footer>
    footer
  </footer>
</DocumentFragment>
`;
